lib/rubocop/cop/rspec/describe_class.rb

```diff
@@ -48,12 +48,29 @@ def on_top_level_describe(node, args)
           return if shared_group?(root_node)
           return if valid_describe?(node)
 
+          unless node.arguments?
+            return add_offense(node, location: :expression,
+                                     message: FIRST_ARGUMENT_REQUIRED_MSG)
+          end
+
           describe_with_metadata(node) do |pairs|
             return if pairs.any?(&method(:rails_metadata?))
           end
 
           add_offense(args.first, location: :expression)
         end
+
+        private
+
+        def valid_describe?(node)
+          return true unless node.arguments? || first_argument_required?
```

Comment:
This doesn't sound right. If first argument isn't required it doesn't check if it's describe with a const

Reply:
The more readable way would be
`return true if !node.arguments? && !first_argument_required?`
and this should be the same as
`return true if !(node.arguments? || first_argument_required?)`
which is the same as
`return true unless node.arguments? || first_argument_required?`

Or am I missing something?

Reply:
`return false if first_argument_required? && node.first_argument.nil?` is not working: when `first_argument_required` is `false` it will go to `describe_with_const?(node)` and this will return false.

The current code is working - event if I add more examples to the case where `FirstArgumentRequired` is set to `true`.

I see that its not that easy to read, but I have a hard time to find a more readable code, that is doing the same.

Reply:
Yup, the fact that the node has arguments enters the check if the argument is `const`, so the code is correct, although counter-intuitive.

It seems that you can just check if node.first_argument.const_type? and not use the node matcher at all